flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -687,6 +684,10 @@ public StreamStatusMaintainer getStreamStatusMaintainer() {
 		return operatorChain.getStreamOutputs();
 	}
 
+	public IntFunction<MailboxExecutor> getMailboxExecutorFactory() {
```

Comment:
What do you think about just having an explicit factory interface instead of using too generic `IntFunction`?